File: main/java/com/marsrovers/MarsRovers.java
package com.marsrovers;

import com.marsrovers.cli.CliComponent;
import com.marsrovers.cli.MainMenu;
import com.marsrovers.control.MissionController;
import com.marsrovers.io.IOHandler;

class MarsRovers {

    private final IOHandler handler;
    private final MissionController controller;
    private final MainMenu menu;

    MarsRovers(IOHandler handler, MissionController controller, MainMenu menu) {
        this.handler = handler;
        this.controller = controller;
        this.menu = menu;
    }

    void menu() {
        handler.write(controller.getStatus());

        CliComponent component = menu.getSelectedCliComponent();

        component.execute();
    }

}

File: main/java/com/marsrovers/MarsRovers.kt
package com.marsrovers

import com.marsrovers.cli.MainMenu
import com.marsrovers.control.MissionController
import com.marsrovers.io.IOHandler

internal class MarsRovers(private val handler: IOHandler, private val controller: MissionController, private val menu: MainMenu) {

    fun menu() {
        handler.write(controller.status)

        val component = menu.selectedCliComponent

        component.execute()
    }

}
